### PR TITLE
feat(terraform): check cognitive services restrict outbound network

### PR DESCRIPTION
**Pull request by tjwald**
# User description
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

Implemented a check for open ai cognitive services in azure to make sure outbound network is restricted.
Implements Prisma Policy Id :"4c4ae367-8609-4b55-ac11-e9d52849658a"

## New/Edited policies (Delete if not relevant)

id = "CKV_AZURE_247"
name = "Ensure that Azure Cognitive Services account hosted with OpenAI is configured with data loss prevention"

* checks that outbound_network_access_restricted is restricted or the fqdns are restricted.

### Description
Check that the cognitive service is either restricted to no access to outbound network, or is restricted to specific fqdns.

### Fix
Either set `outbound_network_access_restricted = true`, or `fqdns = [...]`.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes

---

---

# Generated description

Below is a concise technical summary of the changes proposed in this PR:
<p>Implements a new check for Azure Cognitive Services accounts hosted with OpenAI to ensure data loss prevention. Adds a new class <code>OpenAICognitiveServicesRestrictOutboundNetwork</code> that checks if the outbound network access is restricted or if specific FQDNs are set. Creates test cases to verify the functionality of the new check, including both passing and failing scenarios.</p>
    <table><tr><th>Topic</th><th>Details</th><tr><td><a href=https://baz.co/changes/bridgecrewio/checkov/6919?tool=ast&topic=Test+cases>Test cases</a>
        </td><td>Adds test cases and examples for the new OpenAI Cognitive Services security check<details><summary>Modified files (2)</summary><ul><li>tests/terraform/checks/resource/azure/test_OpenAICognitiveServicesRestrictedOutboundNetwork.py</li>
<li>tests/terraform/checks/resource/azure/example_OpenAICognitiveServicesRestrictOutboundNetwork/main.tf</li></ul></details><details><summary>Latest Contributors(0)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr></table></details></td></tr>
<tr><td><a href=https://baz.co/changes/bridgecrewio/checkov/6919?tool=ast&topic=New+security+check>New security check</a>
        </td><td>Implements a new security check for Azure Cognitive Services OpenAI accounts to ensure data loss prevention<details><summary>Modified files (1)</summary><ul><li>checkov/terraform/checks/resource/azure/OpenAICognitiveServicesRestrictOutboundNetwork.py</li></ul></details><details><summary>Latest Contributors(0)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr></table></details></td></tr></table>This pull request is reviewed by Baz. Join @tjwald and the rest of your team on <a href=https://baz.co/changes/bridgecrewio/checkov/6919?tool=ast>(Baz)</a>.

### CODE REVIEW

File: checkov/terraform/checks/resource/azure/CognitiveServicesRestrictOutboundNetwork.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        categories = (CheckCategories.NETWORKING, )
+        super().__init__(
+            name=name,
+            id=id,
+            categories=categories,
+            supported_resources=supported_resources,
+        )
+
+    def scan_resource_conf(self, conf: Dict[str, List[Any]]) -> CheckResult:
+        if conf.get("kind", [""])[0].lower() != 'openai':
```
Comment (bo156) [inline checkov/terraform/checks/resource/azure/CognitiveServicesRestrictOutboundNetwork.py:21]: do we want this part or should it be for all kinds?

**Review by bo156 — APPROVED** [review]

File: checkov/terraform/checks/resource/azure/CognitiveServicesRestrictOutboundNetwork.py (hunk 1 above)
Comment (tsmithv11) [reply checkov/terraform/checks/resource/azure/CognitiveServicesRestrictOutboundNetwork.py:21]: Just OpenAI. This is a run check we are mirroring

File: checkov/terraform/checks/resource/azure/CognitiveServicesRestrictOutboundNetwork.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from typing import Dict, List, Any
+
+from checkov.common.models.enums import CheckCategories, CheckResult
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+
+
+class CognitiveServicesRestrictOutboundNetwork(BaseResourceCheck):
+    def __init__(self):
+        name = "Ensure that Cognitive Services accounts restrict outbound network traffic."
```
Comment (tsmithv11) [inline checkov/terraform/checks/resource/azure/CognitiveServicesRestrictOutboundNetwork.py:9]:
```suggestion
        name = "Ensure that Azure Cognitive Services account hosted with OpenAI is configured with data loss prevention"
```

**Review by tsmithv11 — CHANGES_REQUESTED** [review]
Looks good! Small title change suggestion

**Review by SteveVaknin — APPROVED** [review]
Nice! 💯

**Pull request merged**